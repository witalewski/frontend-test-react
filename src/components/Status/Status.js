import React from 'react';

const Status = ({ open }) => (
  <div className="status">
    <span className={`status__indicator ${open && 'status__indicator--open'}`}>
      â¬¤
    </span>
    {open ? 'Open Now' : 'Closed'}
  </div>
);

export { Status };
