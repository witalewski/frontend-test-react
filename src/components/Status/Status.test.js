import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MOCK_RESTAURANT from '../../fixtures/restaurant.json';
import { Status } from './Status';

describe('Status', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Status open={!MOCK_RESTAURANT.is_closed} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
