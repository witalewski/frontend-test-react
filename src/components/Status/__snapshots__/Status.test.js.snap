// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status matches snapshot 1`] = `
<div
  className="status"
>
  <span
    className="status__indicator status__indicator--open"
  >
    â¬¤
  </span>
  Open Now
</div>
`;
